WorkflowSwiftUIExperimental/Sources/SwiftUIScreen.swift

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2023 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if canImport(UIKit)
+
+import SwiftUI
+import Workflow
+import WorkflowUI
+
+public protocol SwiftUIScreen: Screen {
+    associatedtype Content: View
+
+    @ViewBuilder
+    static func makeView(model: ObservableValue<Self>) -> Content
+
+    static var isDuplicate: ((Self, Self) -> Bool)? { get }
```

Comment:
Many screens become equatable if we [replace their callback functions with an equatable action sink](https://github.com/square/workflow-swift/pull/225/files#diff-898a1053dd4dcee09d1f6bfe2c1d1b7ab1c01196c96cf8418c54deb4a6fc07ec).

But I'm still not sure that we should be trying to remove duplicates at the `ObservableValue` level rather than rely on SwiftUI's internal comparison of `View` values. I would consider the `isDuplicate` functionality subject to removal in `ObservableValue` and especially in `SwiftUIScreen`.